src/vs/workbench/browser/parts/editor/editorGroupView.ts

```diff
@@ -1014,51 +1014,41 @@ export class EditorGroupView extends Themable implements IEditorGroupView {
 		return this.model.isActive(editor);
 	}
 
-	async selectEditor(editor: EditorInput, active?: boolean): Promise<void> {
-		if (active) {
-			await this.doOpenEditor(editor, { activation: EditorActivation.ACTIVATE }, { selected: true });
-		} else {
-			this.model.selectEditor(editor, active);
-		}
-	}
-
 	async selectEditors(editors: EditorInput[], activeEditor?: EditorInput): Promise<void> {
 		for (const editor of editors) {
-			await this.selectEditor(editor, editor === activeEditor);
+			if (editor === activeEditor) {
+				await this.doOpenEditor(editor, { activation: EditorActivation.ACTIVATE }, { addToSelection: true });
+			} else {
+				this.model.selectEditor(editor);
+			}
 		}
 	}
 
-	async unSelectEditor(editor: EditorInput): Promise<void> {
-		await this.unSelectEditors([editor]);
-	}
-
-	async unSelectEditors(editors: EditorInput[]): Promise<void> {
-		// Check if the active editor is unselected
-		const unselectingActiveEditor = !!editors.find(editor => this.model.isActive(editor));
-		if (unselectingActiveEditor) {
-			editors = editors.filter(editor => !this.model.isActive(editor));
-		}
+	async unselectEditors(editors: EditorInput[]): Promise<void> {
 
 		// Unselect all none active editors
+		let unselectingActiveEditor = false;
 		for (const editor of editors) {
+			if (this.model.isActive(editor)) {
+				unselectingActiveEditor = true;
+				continue;
+			}
 			this.model.unselectEditor(editor);
 		}
```

Comment:
Wouldn't it be better if the model had 1 method to update the selection for N editors? now we iterate over editors, update the selection per editor and each will trigger an event that will update the tabs UI?

Reply:
The reason I only pass one editor is because this is what the model does for every other operation. For example there is also only a `moveEditor` and no `moveEditors`. Also the group event are also only designed to pass 1 editor.

Reply:
Changing to update multiple at once...